airbyte-workers/src/main/java/io/airbyte/workers/WorkerApp.java

```diff
@@ -195,11 +195,15 @@ private void registerSync(final WorkerFactory factory) {
         defaultWorkerConfigs,
         defaultProcessFactory);
 
+    final CheckConnectionActivityImpl checkConnectionActivity =
+        new CheckConnectionActivityImpl(checkWorkerConfigs, checkProcessFactory, secretsHydrator, workspaceRoot, workerEnvironment, logConfigs,
+            jobPersistence, airbyteVersion);
+
     final PersistStateActivityImpl persistStateActivity = new PersistStateActivityImpl(workspaceRoot, configRepository);
 
     final Worker syncWorker = factory.newWorker(TemporalJobType.SYNC.name(), getWorkerOptions(maxWorkers.getMaxSyncWorkers()));
     syncWorker.registerWorkflowImplementationTypes(SyncWorkflowImpl.class);
-    syncWorker.registerActivitiesImplementations(replicationActivity, normalizationActivity, dbtTransformationActivity, persistStateActivity);
+    syncWorker.registerActivitiesImplementations(checkConnectionActivity, replicationActivity, normalizationActivity, dbtTransformationActivity, persistStateActivity);
```

Comment:
here we regsiter the activity in the worker app, which is what runs when you build the docker image proper